src/main/java/com/hillel/items_exchange/service/AdvertisementService.java

```diff
@@ -116,6 +118,20 @@ public void remove(long id) {
         advertisementRepository.flush();
     }
 
+    public void setDefaultImage(Long advertisementId, Long imageId, User owner) throws ClassNotFoundException {
+        final Advertisement advertisement = owner.getAdvertisements().stream()
+                .filter(adv -> adv.getId() == advertisementId)
+                .findFirst().orElseThrow(() -> new ClassNotFoundException(
+                        getExceptionMessageSource("exception.illegal.id")));
+        final Image image = advertisement.getProduct().getImages().stream()
+                .filter(img -> img.getId() == imageId)
+                .findFirst().orElseThrow(() -> new ClassNotFoundException(
+                        getExceptionMessageSource("exception.illegal.id")));
+        advertisement.getProduct().getImages().forEach(img -> img.setDefaultPhoto(false));
+        image.setDefaultPhoto(true);
```

Comment:
You could move such action to your stream into line 127 (after doing filtering) and before that code - set all images false as default.

Reply:
Can't do that. forEach is terminal operation.